File: 1lab/py/1.py
list = [1,3,4,5,6]
i = 4
ans = [-1]
print(ans)
for j in range(i - 1, -1, -1):
    print(i)
    if list[j]<= list[i]:
        ans.append(list[j])
print(ans)

File: 3lab/1.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
    vector <int> nums  = {1,2,2,3, 5};
    int target = 2;
    vector <int> ans;
    int l = 0, r = nums.size() - 1;
    while(r >= l){
        int m = l + (r - l) / 2;
        cout << m;
        if( nums[m] == target){
            ans.push_back(nums[m]);
        }
        else if (nums[m] > target){
            r = m - 1;
        }
        else l = m + 1;
    }
    for (auto x : ans){
        cout << x;
    }
}